internal/httpmodel/new_plan.go

```diff
@@ -2,13 +2,36 @@ package httpmodel
 
 import (
 	"fmt"
+	"time"
 
 	"github.com/cyverse/qms/internal/model"
 )
 
 // Note: the names in the comments may deviate a bit from the actual structure names in order to avoid producing
 // confusing Swagger docs.
 
+// A plan quota default key.
+type PlanQuotaDefaultKey struct {
+	ResourceTypeName string
+	EffectiveDate    int64
+}
```

Comment:
This is a struct to use as a key in a map. The idea is to uniquely identify a plan quota default so that we can check for duplicates before causing a constraint violation in the database, which helps us generate more helpful error messages.